### PR TITLE
Datastore Namespace

### PR DESCRIPTION
**Pull request by blowmage**
Refactor some of the internals for namespace support on queries. Add some documentation as well.

[refs #404]

### CODE REVIEW

File: lib/gcloud/datastore/dataset.rb [hunk 1]
```diff
@@ -239,19 +242,31 @@ def delete *entities_or_keys
...
+      # === Examples
       #
       #   query = Gcloud::Datastore::Query.new.kind("Task").
       #     where("completed", "=", true)
       #   tasks = dataset.run query
       #
-      def run query, opts = {}
-        response = connection.run_query query.to_proto, opts
+      # The query can optionally run within namespace when the +namespace+
+      # option is provided:
```
Comment (quartzmo) [inline lib/gcloud/datastore/dataset.rb]: Maybe: "The query can be scoped to a namespace with the +namespace+ option." ?
And link namespace to external doc if there's a good explanation of it somewhere?

**Comment by quartzmo** [comment]
Other than the typo above ("within namespace"), LGTM and ready to merge.

File: lib/gcloud/datastore/dataset.rb (hunk 1 above)
Comment (blowmage) [inline lib/gcloud/datastore/dataset.rb]: This has been changed.

**Comment by blowmage** [comment]
That isn't a typo! :laughing:

File: lib/gcloud/datastore/dataset.rb (hunk 1 above)
Comment (blowmage) [inline lib/gcloud/datastore/dataset.rb]: I'm not aware of a Google Cloud link that explains namespaces. They are mentioned throughout the Datastore docs, but namespaces and multitenancy are not explicitly defined AFAICT.

**Pull request merged**